tests/functional/kmip/highlevel.spec.js

```diff
@@ -120,4 +122,130 @@ describe('KMIP High Level Driver', () => {
             done();
         });
     });
+
+    it('should retry without kmipHandshake on thales session revoked', async () => {
+        const options = {
+            kmip: {
+                providerName: 'tests',
+                client: {
+                    bucketNameAttributeName: null,
+                    compoundCreateActivate: false,
+                },
+                codec: {},
+                transport: {
+                    pipelineDepth: 8,
+                    tls: {
+                        port: 5696,
+                    },
+                },
+            },
+        };
+        const kmipClient = new KMIPClient(options, TTLVCodec, LoopbackServerTransport);
+        // healthcheck to start the connection and the kmipHandshake
+        void await util.promisify(kmipClient.healthcheck.bind(kmipClient))(logger);
+
+        const kmsKey = await util.promisify(kmipClient.createBucketKey.bind(kmipClient))('mybucket', logger);
+
+        const encrypted = await util.promisify(kmipClient.cipherDataKey.bind(kmipClient))(
+            1, kmsKey, Buffer.from('abc'), logger);
+
+        const reason = 'General Failure';
+        const message = '[NCERRUnauthorizedAccess]: Token has been revoked';
+
+        // the next call should return a thales session revoked error only once
+        const stub = sinon.stub(kmipClient.kmip.transport.channel, '_transform')
+            .onFirstCall()
+            .callsFake((_cipherFunc, _request, cb) => {
+                kmipClient.kmip.transport.channel.stubbedResult = true;
+                cb(null, kmipClient.kmip.transport.channel.errorResponse(reason, message));
+                kmipClient.kmip.transport.channel.stubbedResult = false;
+            })
+            .callThrough();
+        const tlsWriteSpy = sinon.spy(kmipClient.kmip.transport.channel, 'write');
+        const tlsEndSpy = sinon.spy(kmipClient.kmip.transport.channel, 'end');
+        const tlsConnectSpy = sinon.spy(kmipClient.kmip.transport.channel, 'connect');
+
+
+        const encrypted2 = await util.promisify(kmipClient.cipherDataKey.bind(kmipClient))(
```

Comment:
I would rename this from `encrypted2` to `retriedEncrypted` to better reflect intent